File: Clase 18/PYTHON/ejemplo_animales/Vaca.py
from Animal import Animal

class Vaca(Animal):

    def __init__(self, id, nombre, especie, n_estomagos) -> None:
        super().__init__(id, nombre, especie)
        self.n_estomagos = n_estomagos
    
    def hace_muuu(self):
        print(f"{self.nombre} hace MUUUUUUU!!!!")

    # Polimorfismo por inclusion
    def hace_sonido(self):
        self.hace_muuu()

File: Clase 18/JAVA/ejemploAnimales/Vaca.java
package ejemploAnimales;


public class Vaca extends Animal{
    protected int nEstomagos;

    public Vaca(int id, String nombre, String especie, int nEstomagos){
        super(id, nombre, especie);
        this.nEstomagos = nEstomagos;
    }

    public void haceMu(){
        System.out.println(this.nombre + " hace MUUUU!!!!");
    }

    // Polimorfismo Por Inclusion
    @Override
    public void haceSonido(){
        this.haceMu();
    }
}